src/code/PreRender.c

```diff
@@ -254,40 +338,69 @@ void func_800C1F20(PreRender* this, Gfx** gfxp) {
     }
 }
 
-void func_800C1FA4(PreRender* this, Gfx** gfxp) {
+/**
+ * Draws the coverage of the current framebuffer, overwriting it.
+ */
+void PreRender_DrawCoverage16b(PreRender* this, Gfx** gfxp) {
     Gfx* gfx = *gfxp;
 
     gDPPipeSync(gfx++);
+    // Set the blend color to full white and set maximum depth
     gDPSetBlendColor(gfx++, 255, 255, 255, 8);
     gDPSetPrimDepth(gfx++, 0xFFFF, 0xFFFF);
+
+    // Uses G_RM_VISCVG to blit the coverage values to the framebuffer
+    //
+    // G_RM_VISCVG is the following special render mode:
+    //  IM_RD    : Allow read-modify-write operations on the framebuffer
+    //  FORCE_BL : Apply the blender to all pixels rather than just edges
+    //  (G_BL_CLR_IN * G_BL_0 + G_BL_CLR_BL * G_BL_A_MEM) / (G_BL_0 + G_BL_CLR_BL) = G_BL_A_MEM
+    //
+    // G_BL_A_MEM ("memory alpha") is coverage, therefore this blender configuration emits only the coverage
+    // and discards any pixel colors
+    //
+    // Also disables other modes such as alpha compare and texture perspective correction
     gDPSetOtherMode(gfx++,
                     G_AD_DISABLE | G_CD_DISABLE | G_CK_NONE | G_TC_FILT | G_TF_POINT | G_TT_NONE | G_TL_TILE |
                         G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                     G_AC_NONE | G_ZS_PRIM | G_RM_VISCVG | G_RM_VISCVG2);
+    // Set up a scissor with the same dimensions as the framebuffer
     gDPSetScissor(gfx++, G_SC_NON_INTERLACE, 0, 0, this->width, this->height);
+    // Fill rectangle to obtain the coverage values as an RGBA16 image
     gDPFillRectangle(gfx++, 0, 0, this->width, this->height);
     gDPPipeSync(gfx++);
 
     *gfxp = gfx;
 }
 
-void func_800C20B4(PreRender* this, Gfx** gfxp) {
-    func_800C1FA4(this, gfxp);
+/**
+ * Draws the coverage of the current framebuffer `this->fbuf` to an I8 image at `this->cvgSave`. Overwrites
+ * `this->fbuf` in the process.
+ */
+void PreRender_DrawCoverage(PreRender* this, Gfx** gfxp) {
+    PreRender_DrawCoverage16b(this, gfxp);
     LogUtils_CheckNullPointer("this->cvg_save", this->cvgSave, "../PreRender.c", 532);
     if (this->cvgSave != NULL) {
-        func_800C1B24(this, gfxp, this->fbuf, this->cvgSave);
+        PreRender_DrawCoverage8b(this, gfxp, this->fbuf, this->cvgSave);
```

Comment:
So, to recap

- `PreRender_DrawCoverage16b`
  - with `G_RM_VISCVG`
  - overwrites the rgba16 `this->width`*`this->height` framebuffer
  - at `this-fbuf`
  - with its coverage, stored into the color channels (red=green=blue=coverage , over 5 bits)
- `PreRender_DrawCoverage8b`
  - loads the rgba16 `this->width`*`this->height` framebuffer/image (same thing here since coverage doesn't matter anymore)
  - at `this-fbuf`
  - as a ia16 `this->width`*`this->height` image
  - then with `G_RM_OPA_CI2` and combiner=texel0
  - draws it to `cvgSave` as a i8 framebuffer

To summarize further,
- `PreRender_DrawCoverage16b` overwrites `this->fbuf` with its coverage (rgba16)
- `PreRender_DrawCoverage8b` converts the rgba16-stored coverage into i8

The rgba16->ia16->i8 step is interesting, I guess it does `i8 = (r << 3) | (g >> 2) = (r << 3) | (r >> 2)` (r=g=b=coverage)
(which looks a lot like `buffR[i] = (pxIn.r << 3) | (pxIn.r >> 2);` in `PreRender_AntiAliasFilter` and angrylion's rgba16 to rgba32 conversion https://github.com/ata4/angrylion-rdp-plus/blob/6660e93acc94506933238c9a593494684d0d3bde/src/core/n64video/rdp/tmem.c#L2063 )

anyway I would rename the two functions `PreRender_DrawCoverage` is split into like, maybe,
- `PreRender_DrawCoverage16b` -> `PreRender_FbufToCoverage`
- `PreRender_DrawCoverage8b` -> `PreRender_CoverageRgba16ToI8`
(idk really)